crates/oxc_codegen/src/gen.rs

```diff
@@ -1345,7 +1345,7 @@ impl Gen for RegExpLiteral<'_> {
     fn gen(&self, p: &mut Codegen, _ctx: Context) {
         p.add_source_mapping(self.span);
         let last = p.last_byte();
-        let pattern_text = self.regex.pattern.source_text(p.source_text);
+        let pattern_text = self.regex.pattern.to_string();
```

Comment:
Because I'm not calling `CodeGen::build` in `count_char_when_printing`, `self.regex.pattern.source_text(p.source_text)` errors due to `CodeGen::source_text` being an empty string.
```rust
let mut cg = Codegen::new()
    .with_options(CodegenOptions { minify: true, ..CodegenOptions::default() });
cg.print_expression(expr);
cg.into_source_text().len()
```